sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -127,10 +168,96 @@ def main():
         sbom = json.load(f)
 
     base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
-    if "formulation" in sbom:
-        sbom["formulation"].append({"components": base_images_sbom_components})
+    if detect_sbom_format(sbom) == "cyclonedx":
+        if "formulation" in sbom:
+            sbom["formulation"].append({"components": base_images_sbom_components})
+        else:
+            sbom.update({"formulation": [{"components": base_images_sbom_components}]})
     else:
-        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+        root_element1, map1, inverse_map1 = map_relationships(sbom["relationships"])
+
+        packages = []
+        relationships = []
+
+        # Try to calculate root package represeting the container image or directory, which was
+        # used to build the SBOM, based on the relationships maps.
+        # SPDX has relationsship ROOT-ID DESCRIBES MIDDLE-ID which express the fact the SBOM documents
+        # describes container image or directory represented by MIDDLE-ID package.
+        root_package1 = None
+        for r, contains in map1.items():
+            # root package is the one which contains another elements and is in relationship with
+            # the document element where it stand as relatedSpdxElement
+            if contains and inverse_map1.get(r) == root_element1:
+                root_package1 = r
+        # If not root package is found then create one with ID "Uknown" as source for the SBOM
+        # is not known.
+        if not root_package1:
+            root_package1 = "SPDXRef-DocumentRoot-Unknown-"
```

Comment:
Why does `root_package1` have this string value and why does it end with `-`? There seems no code appending other string behind the tailing `-`.

Reply:
Yeah, that's how syft does that. Probably it use naming convention {name}-{version}. I think it can be removed.